src/models/filter.rs

```diff
@@ -193,68 +208,73 @@ impl SourceCodeUnit {
+      }
+    }
+    // If an enclosing node is provided
+    let query = instantiated_filter.enclosing_node();
+    if !query.get_query().is_empty() {
+      // This ensures that the below while loop considers the current node too when checking for filters.
+      // It does not make sense to check for filter if current node is a "leaf" node.
       if node.child_count() > 0 {
-        initial_node = node.child(0).unwrap();
+        node_to_check = node.child(0).unwrap();
+      }
+      if let Some(result) = self._get_match_from_ancestor(rule_store, node_to_check, query) {
+        node_to_check = result;
+      } else {
+        return false;
       }
-      self._check_enclosing_node(filter, rule_store, substitutions, initial_node)
     }
-  }
 
-  fn _check_current_node(
-    &self, filter: Filter, rule_store: &mut RuleStore, substitutions: &HashMap<String, String>,
-    node: Node,
-  ) -> bool {
-    if !self._filter_contains(&filter, rule_store, substitutions, &node) {
+    if !self._filter_contains(
+      &instantiated_filter,
+      rule_store,
+      substitutions,
+      &node_to_check,
+    ) {
       return false;
     }
-    if !self._filter_not_contains(&filter, rule_store, substitutions, &node) {
+    if !self._filter_not_contains(
+      &instantiated_filter,
+      rule_store,
+      substitutions,
+      &node_to_check,
+    ) {
       return false;
     }
     true
   }
 
-  /// This function checks for the contains
-  fn _check_enclosing_node(
-    &self, filter: Filter, rule_store: &mut RuleStore, substitutions: &HashMap<String, String>,
-    initial: Node,
-  ) -> bool {
-    let mut matched_enclosing_node = false;
+  /// This function checks if a query matches with any ancestor of intial
+  fn _get_match_from_ancestor(
+    &self, rule_store: &mut RuleStore, initial: Node, query_str: &TSQuery,
+  ) -> Option<Node> {
     let mut current_node = initial;
     while let Some(parent) = current_node.parent() {
-      let instantiated_filter = filter.instantiate(substitutions);
-      let enclosing_node_query_str = instantiated_filter.enclosing_node();
-      if let Some(p_match) = get_match_for_query(
-        &parent,
-        self.code(),
-        rule_store.query(enclosing_node_query_str),
-        false,
-      ) {
-        matched_enclosing_node = true;
+      if let Some(p_match) =
+        get_match_for_query(&parent, self.code(), rule_store.query(query_str), false)
+      {
         let scope_node = get_node_for_range(
           self.root_node(),
           p_match.range().start_byte,
           p_match.range().end_byte,
         );
-        if !self._filter_contains(&filter, rule_store, substitutions, &scope_node) {
-          return false;
-        }
-        if !self._filter_not_contains(&filter, rule_store, substitutions, &scope_node) {
-          return false;
-        }
-        break;
+        return Some(scope_node);
       }
       current_node = parent;
     }
-    matched_enclosing_node
+    None
```

Comment:
```suggestion
    return None
```
Idk, i thought this would make it more readable